[IntelliJ][ReportedByUser] Uncaught Exception  com.microsoft.azure.toolkit.lib.common.operation.OperationException: setup run configuration for Azure Functions

IntelliJ build version: 2021.3.3 IU-213.7172.25
OS: Mac OS X
JDK: JetBrains s.r.o. 11.0.14.1
Plugin version: 3.66.0-2021.3
Additional Info: None
Parent component: 
```
com.intellij.openapi.wm.impl.IdeRootPane[,0,0,1792x1095,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=com.intellij.util.ui.UIUtil$1@15449392,flags=449,maximumSize=,minimumSize=,preferredSize=]
```
Error message: 
```
IdeaLoggingEvent[message=null, throwable=com.microsoft.azure.toolkit.lib.common.operation.OperationException: setup run configuration for Azure Functions
	at com.microsoft.azure.toolkit.lib.common.operation.AzureOperationAspect.afterThrowing(AzureOperationAspect.java:88)
	at com.microsoft.azure.toolkit.lib.common.operation.AzureOperationAspect.afterThrowing(AzureOperationAspect.java:49)
	at com.microsoft.azure.toolkit.intellij.legacy.function.runner.FunctionRunConfigurationProducer.setupConfigurationFromContext(FunctionRunConfigurationProducer.java:84)
	at com.microsoft.azure.toolkit.intellij.legacy.function.runner.FunctionRunConfigurationProducer.setupConfigurationFromContext(FunctionRunConfigurationProducer.java:32)
Caused by: com.intellij.openapi.progress.ProcessCanceledException
	at com.microsoft.azure.toolkit.intellij.legacy.function.runner.FunctionRunConfigurationProducer.getAzureFunctionMethods(Functio

--- a/PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/function/runner/FunctionRunConfigurationProducer.java
+++ b/PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/function/runner/FunctionRunConfigurationProducer.java
@@ -101,13 +101,17 @@ public class FunctionRunConfigurationProducer extends LazyRunConfigurationProduc
     }
 
     private static Location<PsiMethod> getAzureFunctionMethods(final Location<?> location) {
-        for (Iterator<Location<PsiMethod>> iterator = location.getAncestors(PsiMethod.class, false); iterator.hasNext();) {
-            final Location<PsiMethod> methodLocation = iterator.next();
-            if (FunctionUtils.isFunctionClassAnnotated(methodLocation.getPsiElement())) {
-                return methodLocation;
+        try {
+            for (Iterator<Location<PsiMethod>> iterator = location.getAncestors(PsiMethod.class, false); iterator.hasNext(); ) {
+                final Location<PsiMethod> methodLocation = iterator.next();
+                if (FunctionUtils.isFunctionClassAnnotated(methodLocation.getPsiElement())) {
+                    return methodLocation;
+                }
             }
+            return null;
+        } catch (Throwable e) {
+            return null;
         }
-        return null;
     }
 
     private Module findModule(FunctionRunConfiguration configuration, Module contextModule) {

--- a/PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/survey/CustomerSurveyManager.java
+++ b/PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/survey/CustomerSurveyManager.java
@@ -74,6 +74,9 @@ public class CustomerSurveyManager {
     }
 
     private void showSurveyPopup(final @Nonnull Project project, final ICustomerSurvey survey) {
+        if (project.isDisposed()) {
+            return;
+        }
         final SurveyPopUpDialog popUpDialog = new SurveyPopUpDialog(project, survey, response -> {
             try {
                 if (response == CustomerSurveyResponse.ACCEPT) {

--- a/PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/CreateFunctionAction.java
+++ b/PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/CreateFunctionAction.java
@@ -158,7 +158,7 @@ public class CreateFunctionAction extends CreateElementActionBase {
     @Override
     protected boolean isAvailable(final DataContext dataContext) {
         final Project project = CommonDataKeys.PROJECT.getData(dataContext);
-        if (project == null) {
+        if (project == null || project.isDisposed()) {
             return false;
         }
         final IdeView view = LangDataKeys.IDE_VIEW.getData(dataContext);

